### ISSUE TITLE
Add tweedie distribution

### ISSUE THREAD

**Issue by kellijohnson-NOAA** [nwfsc-assess/VASTWestCoast#43]
Currently only gamma and lognormal are fully supported. Adding support for the newly implemented Tweedie distribution in VAST will allow for a more-complete set of comparisons with {sdmTMB}.

```
ObsModel = c(10, 2)
RhoConfig = c(Beta1 = 3, Beta2 = 0, Epsilon1 = 0, Epsilon2 = 0)
FieldConfig= c(Omega1 = 0, Epsilon1 = 0, Omega2 = "IID", Espilon2 = "IID")
```

# Notes for code changes

* 10 is for the Tweedie distribution
* Turn off random-effects for temporal variation in the first intercept `RhoConfig["Beta1"]=3` as documented in [`VAST::check_fit()`](https://github.com/James-Thorson-NOAA/VAST/blob/339db318558073ce36098ff56d1e745513313ae7/R/check_fit.R)
* Using 0 for all ...1 parameters turns them off for the first model, which makes sense because there is only one model with the Tweedie distribution.

# [Definition/Explanation of Tweedie](https://www.statisticshowto.com/tweedie-distribution/)

```
The Tweedie distribution is a special case of an exponential distribution.
It can have a cluster of data items at zero (called a “point mass”),
which is particularly useful for modeling claims in the insurance industry,
in medical/genomic testing, or
anywhere else there is a mixture of zeros and non-negative data points.
Basically, if you see a histogram with a spike at zero,
it’s a possible candidate to be fitted to a Tweedie model.
```

**Comment by James-Thorson-NOAA** [comment]
In case you're interested in more theory for the Tweedie, see e.g., https://doi.org/10.1002/ecy.3637, and I also have a paper in review (with Tim Miller and Brian Stock) about how it arises naturally for comp-data via the hierarchical expansion of counted animals.  obviously the Poisson-link delta model is still a more flexible generalization that requires little extra computation time.

**Comment by kellijohnson-NOAA** [comment]
Thanks Jim!
